src/main/java/org/traccar/directions/timeDistance/LocationIQTimeDistance.java

```diff
@@ -0,0 +1,54 @@
+package org.traccar.directions.timeDistance;
+
+import org.traccar.Context;
+
+import javax.json.JsonObject;
+import javax.ws.rs.client.Invocation;
+import java.util.List;
+import java.util.Objects;
+import java.util.StringJoiner;
+
+public class LocationIQTimeDistance extends JsonTimeDistance {
+    public LocationIQTimeDistance(String url, String key) {
+        super(url, key);
+    }
+
+    @Override
+    public JsonObject getTimeDistanceResponse(String url, String key, TimeDistanceRequest timeDistanceRequest) {
+        if (url == null) {
+            url = "https://us1.locationiq.com/v1/matrix/driving/";
+        }
+        String metrics = String.join(",", timeDistanceRequest.getMetrics());
+
+        StringJoiner locationsStringJoiner = new StringJoiner(";");
+        for (List<Double> coordinates : timeDistanceRequest.getLocations()) {
+            StringJoiner locationStringJoiner = new StringJoiner(",");
```

Comment:
This still doesn't make sense to me. Why simple `coordinates.join` doesn't work here?

Reply:
String.join only accepts List<String>.